internal/commands/run.go

```diff
@@ -145,7 +140,14 @@ func run(cmd *cobra.Command, args []string) error {
 	)
 
 	// Container backend
-	executorOpts = append(executorOpts, executor.WithContainerBackend(backend))
+	if containerBackend != containerbackend.BackendAuto {
```

Comment:
Logic here and down in the PR seems to me a bit hard to understand. Maybe because container backends can be both strings and real types.

While I was reviewing the change I ended up looking a bit too much and tried a similar approach in #525 but I think it's a bit easier to wrap your head around. What do you think?